gateway-ha/src/main/java/io/trino/gateway/ha/router/BackendStateManager.java

```diff
@@ -31,27 +29,14 @@ public BackendStateManager()
         this.clusterStats = new HashMap<>();
     }
 
-    public BackendState getBackendState(ProxyBackendConfiguration backend)
+    public ClusterStats getBackendState(ProxyBackendConfiguration backend)
     {
         String name = backend.getName();
-        ClusterStats stats = clusterStats.getOrDefault(backend.getName(), ClusterStats.builder(name).build());
-        Map<String, Integer> state = new HashMap<>();
-        state.put("QUEUED", stats.queuedQueryCount());
-        state.put("RUNNING", stats.runningQueryCount());
-        return new BackendState(name, state);
+        return SerializationUtils.clone(clusterStats.getOrDefault(name, ClusterStats.builder(name).build()));
```

Comment:
Why do we need to clone this object?

Reply:
This is to retain the same behavior of this line https://github.com/trinodb/trino-gateway/pull/341/files#diff-ee8580184f392f093ac0b7163a819db0db2dae61e70b7b11cc860bc57d603336L54

Not sure why it was copied in the first place

Reply:
Please just return `clusterStats.getOrDefault(name, ClusterStats.builder(name).build())`

Reply:
done